KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,367 @@
+
+            del kls_list[i]
+        else:
+            i += 1
+    kls_list.append("Limit Relation")
+    for i in kls_list:
+        add = True
+        for j in kls_list_chap:
+            if i == j:
+                add = False
+        if add:
+            kls_list_chap.append(i)
+    return kls_list_chap
 
 
-def prepareForPDF(chap):
-    footmiscIndex = 0
+def fix_chapter_sort(kls, chap, word, sortloc):
+    """
+    This function sorts through the input files and identifies and places sections from the addendum after their
+    respective subsections in the chapter.
+
+    :param kls: The addendum where the sections to be inserted are found.
+    :param chap: The destination of inserted sections.
+    :param word: The keyword that is being processed.
+    :param sortloc: The location in the list of words, of the word being sorted.
+    :return: This function outputs the processed chapter into a larger function for further processing.
+    """
+    hyper_headers_chap = []
+    hyper_subs_chap = []
+
+    sep1 = 0
+    name_chap = word.lower()
+    if name_chap == "orthogonality":
+        sep1 = 1
+    elif name_chap == "special value":
+        sep1 = 2
+
+    khyper_header_chap = []
+    k_hyper_sub_chap = []
+    index = 0
+
+    for i in kls:
+        index += 1
+        line = str(i)
+        if "\\subsection" in line:
+            temp = line[line.find(" ", 12)+1: line.find("}", 12)]  # get just the name (like mathpeople)
+
+        if sep1 == 0:
+            if name_chap in line.lower() and kls.index(i) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line):
+                for i in klsaddparas:
+                    if index < i:
+                        klsloc = klsaddparas.index(i)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+        elif sep1 == 1:
+            if name_chap in line.lower() and kls.index(i) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line) and "orthogonality relation" not in line.lower():
+                for i in klsaddparas:
+                    if index < i:
+                        klsloc = klsaddparas.index(i)
+                        break
+                t = ''.join(kls[index: klsaddparas[klsloc]])
+                k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+                khyper_header_chap.append(temp)  # append the name of subsection
+        elif sep1 == 2:
+            if name_chap in line.lower() and kls.index(i) > 313 and ("\\paragraph{" in line or
+            "\\subsubsection*{" in line):
+                for i in klsaddparas:
+                    if index < i:
+                        klsloc = klsaddparas.index(i)
```

Comment:
Instead of using `.index()`, maybe use `enumerate` at the top. Also, the variable which is not an index should not be named `i`.